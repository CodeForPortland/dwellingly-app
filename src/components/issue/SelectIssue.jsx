import React from 'react';
import PropTypes from 'prop-types';
import Input from '../input/Input';

const SelectIssue = ({ issueOptions, activeStep, issue, onChange }) => (
  <fieldset className={activeStep === 'issue' ? 'active' : ''}>
    <div className="message message--light">
      <p>
        <strong>Step 1: Select Issue</strong>
      </p>
    </div>
    {issueOptions.map(option => {
      const key = Object.keys(option)[0];
      return (
        <Input
          id={`settings-issue-${key}`}
          key={key}
          label={option[key]}
          model={issue}
          name="issue"
          onChange={onChange}
          type="radio"
          value={option[key]}
        />
      );
    })}
  </fieldset>
);

SelectIssue.propTypes = {
  activeStep: PropTypes.string.isRequired,
  issueOptions: PropTypes.arrayOf(PropTypes.object),
  issue: PropTypes.string,
  onChange: PropTypes.func.isRequired
};

SelectIssue.defaultProps = {
  issue: undefined,
  issueOptions: undefined
};

export default SelectIssue;
