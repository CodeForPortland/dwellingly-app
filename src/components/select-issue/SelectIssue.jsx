import React from 'react';
import PropTypes from 'prop-types';
import Input from '../input/Input';

const SelectIssue = ({ issueOptions, issue, onChange }) => (
  <fieldset>
    <div className="message message--light">
      <p>
        <strong>Step 1: Select Issue</strong>
      </p>
    </div>
    {issueOptions.map(option => {
      const key = Object.keys(option)[0];
      return (
        <Input
          id={`settings-issue-${key}`}
          key={key}
          label={option[key]}
          model={issue}
          name="issue"
          onChange={onChange}
          type="radio"
          value={option[key]}
          />
      );
    })}
  </fieldset>
)

SelectIssue.propTypes = {
  issueOptions: PropTypes.arrayOf(
    PropTypes.shape(
      PropTypes.string
    )
  ),
  issue: PropTypes.string,
  onChange: PropTypes.func.isRequired
}

SelectIssue.defaultProps = {
  issue: undefined,
  issueOptions: undefined
}

export default SelectIssue;
