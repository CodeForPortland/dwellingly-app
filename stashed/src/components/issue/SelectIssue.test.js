import React from 'react';
import renderer from 'react-test-renderer';
import SelectIssue from './SelectIssue';

describe('SelectIssue Component', () => {
  it('renders correctly when issueOptions prop exists', () => {
    const props = {
      issueOptions: [{}],
    }
    const SelectIssueComponent = renderer.create(<SelectIssue {...props}/>).toJSON();
    expect(SelectIssueComponent).toMatchSnapshot();
  });
});
