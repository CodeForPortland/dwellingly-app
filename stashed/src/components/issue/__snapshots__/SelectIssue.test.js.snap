// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectIssue Component renders correctly when issueOptions prop exists 1`] = `
<fieldset
  className=""
>
  <div
    className="message message--light"
  >
    <p>
      <strong>
        Step 1: Select Issue
      </strong>
    </p>
  </div>
  <div
    className="input inline-input"
  >
    <div
      className="inline-input__radio "
    >
      <label
        htmlFor="settings-issue-undefined"
      >
        <input
          checked={true}
          id="settings-issue-undefined"
          name="issue"
          type="radio"
        />
        <span
          className="inline-input__radio-btn"
        />
      </label>
    </div>
  </div>
</fieldset>
`;
